dsgrid/config/representative_period_time_dimension_config.py

```diff
@@ -104,78 +107,106 @@ def get_timestamp_load_data_columns():
 
-OneWeekPerMonthByHourType = namedtuple(
-    "OneWeekPerMonthByHourType", ["month", "day_of_week", "hour"]
-)
+        Returns
+        -------
+        list
+            List of tuples of columns representing time in the load_data table.
+
+        """
 
 
 class OneWeekPerMonthByHourHandler(RepresentativeTimeFormatHandlerBase):
     """Handler for format with hourly data that includes one week per month."""
 
     def check_dataset_time_consistency(self, expected_timestamps, load_data_df):
-        assert False, "not supported yet"
-        # actual_timestamps = [
-        #    OneWeekPerMonthByHourType(month=x.month, day_of_week=x.day_of_week, hour=x.hour)
-        #    for x in load_data_df.select("month", "day_of_week", "hour")
-        #    .distinct()
-        #    .sort("month", "day_of_week", "hour")
-        #    .collect()
-        # ]
-        # if expected_timestamps != actual_timestamps:
-        #    mismatch = sorted(
-        #        set(expected_timestamps).symmetric_difference(set(actual_timestamps))
-        #    )
-        #    raise DSGInvalidDataset(
-        #        f"load_data timestamps do not match expected times. mismatch={mismatch}"
-        #    )
-
-        # timestamps_by_id = (
-        #    load_data_df.select("month", "day_of_week", "hour", "id")
-        #    .groupby("id")
-        #    .agg(F.countDistinct("month", "day_of_week", "hour").alias("distinct_timestamps"))
-        # )
-        # lengths = timestamps_by_id.select("distinct_timestamps").distinct()
-        # count = lengths.count()
-        # if count != 1:
-        #    raise DSGInvalidDataset(
-        #        f"All time arrays must have the same times: unique times={count}"
-        #    )
-
-        # expected_len = len(expected_timestamps)
-        # actual = lengths.collect()[0].distinct_timestamps
-        # if actual != expected_len:
-        #    raise DSGInvalidDataset(
-        #        f"Length of time arrays is incorrect: actual={actual} expected={expected_len}"
-        #    )
+        logger.info("Check OneWeekPerMonthByHourHandler dataset time consistency.")
+        actual_timestamps = [
+            OneWeekPerMonthByHourType(month=x.month, day_of_week=x.day_of_week, hour=x.hour)
+            for x in load_data_df.select("month", "day_of_week", "hour")
+            .distinct()
+            .sort("month", "day_of_week", "hour")
+            .collect()
+        ]
+        if expected_timestamps != actual_timestamps:
+            mismatch = sorted(
+                set(expected_timestamps).symmetric_difference(set(actual_timestamps))
+            )
+            raise DSGInvalidDataset(
+                f"load_data timestamps do not match expected times. mismatch={mismatch}"
+            )
+        logger.info("Verified that expected_timestamps equal actual_timestamps")
+
+        lengths = (
+            load_data_df.select("month", "day_of_week", "hour", "id")
```

Comment:
Presence of `id` implies it's for Standard SchemaHandler only

Reply:
Many check instead df.count group by distinct_timestamps is the same for each timestamp. I.e. lengths = df.groupBy("distinct_timestamps").count().